### DISCUSSION TITLE
Filtering strategy configurable

### DISCUSSION THREAD

**Discussion by Gentleman03** [RafaelGB/obsidian-db-folder#225] [Ideas]
**Describe the bug**
I have noticed that searching doesn't work right in 2.0.0 beta 2. With version 1.8.2 if I search for books with autor "XYZ" it only displays me files with this author - correct behaviour. But in 2.0.0 Beta 2 it displays me books with author "XYZ" and some (sometimes quite a lot of) other books that doesn't have this author assigned.

**To Reproduce**
Just create few files, add column with specific words (in my case book authors) to this column and type one of them into the search. 

**Expected behavior**
You should see only files with certain word assigned to them but you will see also some files that doesn't contain this word. 

**Desktop (please complete the following information):**
 - Windows 10 Pro

**Comment by RafaelGB** [comment]
could be a problem with the path of the note. You can see only the note name but the search reads the entire path. Will be reviewed

**Comment by RafaelGB** [comment]
After an interesting investigation, the simplest way to obtain a exigent filter is with the native "includesString"
You can find the documentation [here](https://tanstack.com/table/v8/docs/api/features/filters#filter-functions)

I was using a custom one where you do not filter the rows on the exact term of the word, but assign a value based on the similarity and order it according to that data

Was the official globalFilterFn example
```ts
const fuzzyFilter: FilterFn<RowDataType> = (row, columnId, value, addMeta) => {
    // Rank the item
    const itemRank = rankItem(row.getValue(columnId), value);

    // Store the itemRank info
    addMeta({
        itemRank,
    });

    // Return if the item should be filtered in/out
    return itemRank.passed;
};
```
In the future we could make the filtering strategy configurable. So I will convert this issue into a discussion. Community I want your opinion!

**Reply by Gentleman03** [reply]
Well I don't exactly understand what do you mean but in version 2.0.1 it works correctly like in 1.8.2. I need to get familiar with documentation. Thank you very much for taking your time and checking this. 
It's such an awesome plugin.

**Reply by RafaelGB** [reply]
anytime!
About the configuration are different ways to engage the filter, I dont know if that apport value
